### DISCUSSION TITLE
How can we decode long value?

### DISCUSSION THREAD

**Discussion by muhammadjehanzaib** [Cyfrin/foundry-full-course-cu#2429] [Q&A] [answered]
i called view function by using 

> cast call 0x5fbdb2315678afecb367f032d93f642f64180aa3 "name()" --account smartkey

and got this big hex value in return

> 0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000005446f676965000000000000000000000000000000000000000000000000000000

now, i want to read this value.
i have tried multiple commands to read it but i doesn't work. like

cast --to-dec 0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000005446f676965000000000000000000000000000000000000000000000000000000

> Error: 
> the value is too large to fit the target type

>  cast abi-decode "string" 0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000005446f676965000000000000000000000000000000000000000000000000000000
> Error: 
> could not parse function signature

**Comment by EngrPips** [comment, marked as answer]
Hello @muhammadjehanzaib, This is how you do it 

```js
cast --to-ascii 0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000005446f676965000000000000000000000000000000000000000000000000000000
 Dogie

```
